tests/junit.py

```diff
@@ -167,8 +168,22 @@ def run(test, backends):
         output = (os.path.join('build', args.test + '.junit')
                   if args.output is None
                   else args.output)
-        with open(output, 'w') as fd:
-            TestSuite.to_file(fd, [result])
+
+        if os.path.exists(output):
```

Comment:
I don't like this race condition. Can we not write different files?

Reply:
I'm not sure under what use case we'd see a race condition? A single invocation of `make junit -j` will only ever write once to a `.junit` file with the results for all backends. But currently if we call `make junit -j` a second time (once for all CPU backend, then again for all GPU backends), then the results from the first invocation are overwritten. This change just appends the new data to the old results.

Do we want to support two independent processes simultaneously calling `make junit` in the same folder at the same time? Wouldn't we always use `make junit -j` in that case?

Reply:
We could certainly instead just append a random value into the filename if that is preferred, it just seemed like a lot of extra clutter.

Reply:
I'd rather just make a unique file name for now. Perhaps pass an env variable to put into the file name. I'd prefer that repeated invocation be idempotent and not leave stale results. Like if you run, fix a bug, and re-run, you should see a clean result. I know that workflow isn't relevant to CI, but minimizing surprise is good.

Reply:
For example, `make junit -j JUNIT_BATCH=cuda` would generate `result.cuda.junit`.